ruby/GNUmakefile

```diff
@@ -35,6 +35,11 @@ ifeq ($(ruby),)
     $(eval $(call pkg_check,libpulse,audio.pulseaudiosimple))
     ruby += input.xlib
     $(eval $(call pkg_check,sdl2,input.sdl))
+  else ifeq ($(platform),haiku)
+    ruby += video.glx video.glx2
+    ruby += audio.oss
+    $(eval $(call pkg_check,openal,audio.openal))
+    $(eval $(call pkg_check,sdl2,input.sdl))
```

Comment:
If you can't build the bsnes desktop UI due to lack of GTK+ or equivalent, have you actually tested this code?

If Haiku doesn't have GTK+, does it have Qt?

Reply:
I was quite positive this would not impact the libretro core but wanted to play it safe with libs I knew existed in Haiku.
Haiku has QT, though, I had not realized bsnes offered a QT interface. I can look into getting this to build.

Reply:
Quick update : working on this, but since Haiku does not use X11 and has its own display things are a little bit hairier than I hoped for, as I am not really a developer :D but things are moving forward slowly.

Reply:
Okay so here's the situation : building standalone bsnes on Haiku will require substantial code changes that I don't feel comfortable doing.
The QT flavor of hiro seems to be building fine, and is not the issue.
The issue is that Haiku does not have a proper shm.h implementation, which won't allow building the XShm or XVideo outputs even if I install X11 libs. 
It also does not implement the various glX functions bsnes uses (even though most are defined in the GLEW headers, they are not present in the Mesa libs in the system) which means no OpenGL output.
I suggest we stick with the libretro core right now. I will submit the other changes shortly.